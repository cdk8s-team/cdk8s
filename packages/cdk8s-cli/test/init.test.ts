import { execSync } from 'child_process';
import * as os from 'os';
import * as fs from 'fs';
import * as path from 'path';

const cli = path.join(__dirname, '..', 'bin', 'cdk8s');

test('typescript-app', () => {
  withTempDir('heythere', () => {
    shell(`${cli} init typescript-app`);
    shell(`yarn synth`);
    expect(fs.readFileSync('dist/heythere.k8s.yaml', 'utf-8')).toMatchSnapshot();
  });
});

function shell(command: string) {
  execSync(command, { stdio: 'inherit', env: { ...process.env, CLI_TEST: '1' }});
}

function withTempDir(dirname: string, closure: () => void) {
  const prevdir = process.cwd();
  const parent = fs.mkdtempSync(path.join(os.tmpdir(), 'cdk8s-init-test.'));
  const workdir = path.join(parent, dirname);
  fs.mkdirSync(workdir);
  try {
    process.chdir(workdir);
    closure();
  } finally {
    process.chdir(prevdir);
    fs.rmdirSync(parent, { recursive: true });
  }
}