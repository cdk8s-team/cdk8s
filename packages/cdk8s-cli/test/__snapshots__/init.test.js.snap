// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript-app 1`] = `
"spec:
  containers:
    - name: hello-kubernetes
      image: paulbouwer/hello-kubernetes:1.7
      ports:
        - containerPort: 8080
kind: Pod
apiVersion: v1
metadata:
  name: heythere-mypod-66ae4712
"
`;
